test/unit/cmap/auth/mongodb_oidc/azure_machine_workflow.test.ts

```diff
@@ -0,0 +1,24 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { AzureMachineWorkflow, Connection, MongoCredentials } from '../../../../mongodb';
+
+describe('AzureMachineFlow', function () {
+  describe('#execute', function () {
+    const workflow = new AzureMachineWorkflow();
+
+    context('when TOKEN_RESOURCE is not set', function () {
+      const connection = sinon.createStubInstance(Connection);
+      const credentials = sinon.createStubInstance(MongoCredentials);
+
+      it('throws an error', async function () {
+        try {
```

Comment:
A little confused here, are you looking for the execute to throw an error with "TOKEN_RESOURCE" in the message or not? the expect.fail cannot contain the string you are expecting to only come out of workflow.execute if so. 

Personally, I would remove the control flow risk from the test, but any solution works here.

```js
const error = await workflow.execute(connection, credentials).catch(error => error);
expect(error)...
```

Reply:
I've updated to the suggestion.